### PR TITLE
Ntfy icon support

### LINKED ISSUE #2135: Add icon support to ntfy notification provider
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

Other

### 🔖 Feature description

Add icon support to the ntfy notification provider (icon support has been added to ntfy with a release on 2022-09-27).

### ✔️ Solution

There's just another parameter to have in the notification providers' settings and the http request sent.

### ❓ Alternatives

none

### 📝 Additional Context

_No response_

### PR DESCRIPTION
**Pull request by cmeis**
Tick the checkbox if you understand [x]: 
- [x] I have read and understand the pull request rules.

# Description

Fixes #2135 by adding support for icon in the ntfy notification provider

## Type of change

- Other

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [ ] I have commented my code, particularly in hard-to-understand areas
  (including JSDoc for methods)
- [x] My changes generate no new warnings
- [ ] My code needed automated testing. I have added them (this is optional task)

### CODE REVIEW

File: src/languages/en.js [hunk 1]
```diff
@@ -582,4 +582,5 @@ export default {
     goAlert: "GoAlert",
     backupOutdatedWarning: "Deprecated: Since a lot of features added and this backup feature is a bit unmaintained, it cannot generate or restore a complete backup.",
     backupRecommend: "Please backup the volume or the data folder (./data/) directly instead.",
+    IconUrl: "Icon-URL",
```
Comment (Computroniks) [inline src/languages/en.js:585]: Should this be hyphenated?
```suggestion
    IconUrl: "Icon URL",
```

Comment (cmeis) [reply src/languages/en.js:585]: Well, it probably shouldn't for English, that's my German roots coming through ;-) Thanks!

File: src/components/notifications/Ntfy.vue [hunk 2]
```diff
@@ -16,6 +16,11 @@
         <label for="ntfy-priority" class="form-label">{{ $t("Priority") }}</label>
         <input id="ntfy-priority" v-model="$parent.notification.ntfyPriority" type="number" class="form-control" required min="1" max="5" step="1">
     </div>
+
+    <div class="mb-3">
+        <label for="ntfy-icon" class="form-label">{{ $t("IconUrl") }}</label>
+        <input id="ntfy-icon" v-model="$parent.notification.ntfyIcon" type="text" class="form-control" required>
```
Comment (louislam) [inline src/components/notifications/Ntfy.vue:22]: Need required?

File: server/notification-providers/ntfy.js [hunk 3]
```diff
@@ -13,6 +13,7 @@ class Ntfy extends NotificationProvider {
                 "message": msg,
                 "priority": notification.ntfyPriority || 4,
                 "title": "Uptime-Kuma",
+                "icon": notification.ntfyIcon || "",
```
Comment (louislam) [inline server/notification-providers/ntfy.js:16]: > X-Icon | Icon | URL to use as notification icon

The api doc do not say anything about empty string, I think it is not a proper handling.

File: src/components/notifications/Ntfy.vue (hunk 2 above)
Comment (cmeis) [reply src/components/notifications/Ntfy.vue]: Well. No. Removed the required attributed.

File: server/notification-providers/ntfy.js (hunk 3 above)
Comment (cmeis) [reply server/notification-providers/ntfy.js]: The docs aren't really clear there. In current ntfy releases this works, but I changed the notification provider to only add the Icon header if there actually is an icon.

**Pull request merged**